cudamapper/src/index_gpu.cuh

```diff
@@ -130,13 +132,22 @@ public:
     /// \return length of the longest read in this index
     position_in_read_t number_of_basepairs_in_longest_read() const override;
 
+    /// \brief checks if index is ready to be used, index might not be ready if its creation is asynchronous
+    /// \return whether the index is ready to be used
+    bool is_ready() const override;
+
+    /// \brief if is_ready() is true returns immediately, blocks until it becomes ready otherwise
+    void wait_to_be_ready() const override;
```

Comment:
Does this need to be `const`?

Reply:
For now it does not have to be `const`. I was thinking a lot about whether object becoming ready means a change of state, but for now can leave make is non-`const`
done